iis/datadog_checks/iis/iis.py

```diff
@@ -55,62 +55,71 @@ def get_iishost(self, instance):
-
-                for sitename, val in iteritems(vals):
-                    tags = []
-                    if key in self._tags:
-                        tags = list(self._tags[key])
-                    tags.append(self.get_iishost(instance))
-
-                    try:
-                        if not counter.is_single_instance():
-                            if sites and sitename != TOTAL_SITE and sitename not in sites:
-                                continue
-                            tags.append("site:{}".format(self.normalize(sitename)))
-                    except Exception as e:
-                        self.log.error("Caught exception {} setting tags".format(e))
+                site_values = counter.get_all_values()
+            except Exception as e:
+                self.log.error("Failed to get_all_values {} {}: {}".format(inst_name, dd_name, e))
+                continue
+            try:
+                for site_name, site_uptime in iteritems(site_values):
+                    if not counter.is_single_instance():
+                        if site_name != TOTAL_SITE and site_name not in sites:
+                            continue
+                    tags = self._get_site_tags(instance_hash, instance, site_name)
 
                     try:
-                        metric_func(dd_name, val, tags)
+                        metric_func(dd_name, site_uptime, tags)
                     except Exception as e:
-                        self.log.error("Error in metric_func: {} {} {}".format(dd_name, val, e))
+                        self.log.error("Error in metric_func: {} {} {}".format(dd_name, site_uptime, e))
 
                     if dd_name == "iis.uptime":
-                        uptime = int(val)
-                        status = AgentCheck.CRITICAL if uptime == 0 else AgentCheck.OK
-                        self.service_check(self.SERVICE_CHECK, status, tags)
-                        if sitename in expected_sites:
-                            self.log.debug("Removing {!r} from expected sites".format(sitename))
-                            expected_sites.remove(sitename)
+                        self._report_uptime(site_uptime, tags)
+                        if site_name in expected_sites:
+                            self.log.debug("Removing {!r} from expected sites".format(site_name))
+                            expected_sites.remove(site_name)
                         else:
-                            self.log.warning("Site {!r} not in expected_sites".format(sitename))
+                            self.log.warning("Site {!r} not in expected_sites".format(site_name))
 
             except Exception as e:
                 # don't give up on all of the metrics because one failed
                 self.log.error("IIS Failed to get metric data for {} {}: {}".format(inst_name, dd_name, e))
 
-        for site in expected_sites:
+        self._report_unavailable_sites(expected_sites, instance_hash, instance)
```

Comment:
Same here.

Reply:
I just felt the check method was too long to easily understand